### PR TITLE
Add `jsdoc` bundle package

### PR DESCRIPTION
**Pull request by tomalec**
### Changes proposed in this Pull Request:

Addresses some of the concerns from https://github.com/woocommerce/grow/pull/6#pullrequestreview-984688504

Bundles multiple commonly used `jsdoc` plugins, so the extensions repo would need only this one dependency.
Exports a binary, that wraps commonly used `jsdoc` CLI arguments, for easier use.

### Detailed test instructions:

See example usage at https://github.com/woocommerce/google-listings-and-ads/compare/develop...tweak/bundled-jsdoc
And the installation & usage instructions at https://github.com/woocommerce/grow/blob/add/jsdoc-bundle/packages/js/jsdoc/README.md 

1. `npm i --save-dev https://gitpkg.now.sh/woocommerce/grow/packages/js/jsdoc?add/jsdoc-bundle`
2. Create a `TRACKING.md` file
   ```md 
   # Usage Tracking
   Some nice general description.
   <woocommerce-grow-tracking-jsdoc></woocommerce-grow-tracking-jsdoc>
   ```
4. Create a `.jsdocrc.json`
   ```json
   {
      "plugins": [
         "woocommerce-grow-jsdoc/tilde-alias",
         "woocommerce-grow-tracking-jsdoc/fires-description",
         "jsdoc-plugin-typescript",
         "jsdoc-advanced-types-plugin",
         "jsdoc-plugin-intersection"
      ],
      "typescript": {
         "moduleRoot": "js/src"
      }
   }
   ```
6. `npx woocommerce-grow-jsdoc`

### Changelog entry

> Add - `-jsdoc` bundle package

### CODE REVIEW

File: packages/js/storybook/bin/storybook.js [hunk 1]
```diff
@@ -3,11 +3,10 @@
...
  */
 import { argv } from 'process';
+import shell from 'shelljs';
+import path from 'path';
 
-/**
- * Internal dependencies
- */
-import { shell } from '../utils.js'
+process.env.PATH += ( path.delimiter + path.join( process.cwd(), 'node_modules', '.bin') );
```
Comment (puntope) [inline packages/js/storybook/bin/storybook.js:9]: This line is repeated over all the bin files, so I guess could be reuse it in `utils.js` kinda file.

**Comment by puntope** [comment]
⚠️ ❓  

When I ran the installation I got this error:

```
➜  npm i --save-dev "https://gitpkg.now.sh/woocommerce/grow/packages/js/jsdoc?add/jsdoc-bundle"
npm ERR! code ENOLOCAL
npm ERR! Could not install from "node_modules/tracking-jsdoc" as it does not contain a package.json file.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/miguelperezpellicer/.npm/_logs/2022-06-07T12_58_34_945Z-debug.log
```

File: packages/js/storybook/bin/storybook.js (hunk 1 above)
Comment (tomalec) [reply packages/js/storybook/bin/storybook.js:9]: In all 2 ;) We cannot.
as npm fails to resolve shelljs dependency in a parent folder without its own package.json
https://github.com/woocommerce/grow/pull/10/commits/99141963a638199522a213fdda065f05b7847a9a

Plus it's single line
```js
/**
 * Internal dependencies
 */
import { shell } from '../../utils.js'
```
vs.
```js
import shell from 'shelljs';
import path from 'path';

process.env.PATH += ( path.delimiter + path.join( process.cwd(), 'node_modules', '.bin') );
```

So not a big difference in LOC, and a bit more explicit, that it's just the original `shelljs`

**Comment by tomalec** [comment]
> When I ran the installation I got this error:

Thanks, it was a leftover from my local testing with a relative filesystem reference. Fixed in https://github.com/woocommerce/grow/pull/10/commits/8b88bfdee0d03cf5bd1539cb2856e3dc7bcfa2e2

**Comment by tomalec** [comment]
@puntope would you mind another look?

**Review by puntope — APPROVED** [review]
✅  LGTM

I was able to install the plugin and generate the tracking docs 

Thanks ❤️

**Pull request merged**